diff --git a/configure.ac b/configure.ac
index 8469ea2..fb2a23b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -99,6 +99,15 @@ PKG_CHECK_MODULES([aom], [aom], [
 AM_CONDITIONAL([HAVE_AOM], [test "x$have_aom" = "xyes"])
 AC_SUBST([have_aom])
 
+PKG_CHECK_MODULES([rav1e], [rav1e], [
+    AC_DEFINE([HAVE_RAV1E], [1], [Whether rav1e was found.])
+    AC_SUBST(rav1e_CFLAGS)
+    AC_SUBST(rav1e_LIBS)
+    have_rav1e="yes"
+], [have_rav1e="no"])
+AM_CONDITIONAL([HAVE_RAV1E], [test "x$have_rav1e" = "xyes"])
+AC_SUBST([have_rav1e])
+
 PKG_CHECK_MODULES([libde265], [libde265], [
     AC_DEFINE([HAVE_LIBDE265], [1], [Whether libde265 was found.])
     AC_SUBST(libde265_CFLAGS)
@@ -194,10 +203,16 @@ AC_ARG_ENABLE([local_rav1e], AS_HELP_STRING([--enable-local-rav1e],
 if eval "test x$enable_local_rav1e != xno"; then
     AC_MSG_NOTICE([Enable compiling with local rav1e])
     AC_DEFINE([HAVE_RAV1E], [1], [Whether rav1e plugin has been enabled.])
-fi
-AM_CONDITIONAL([ENABLE_LOCAL_RAV1E], [test "x$enable_local_rav1e" != "xno"])
-AC_SUBST(ENABLE_LOCAL_RAV1E)
 
+    have_rav1e="yes"
+    HAVE_RAV1E=1
+    rav1e_CFLAGS="-I../third-party/rav1e/target/release/include"
+    rav1e_LIBS="../third-party/rav1e/target/release/librav1e.a"
+
+    AC_SUBST(have_rav1e)
+    AC_SUBST(rav1e_CFLAGS)
+    AC_SUBST(rav1e_LIBS)
+fi
 
 AC_ARG_ENABLE([libfuzzer], AS_HELP_STRING([--enable-libfuzzer],
     [Enable fuzzing with libFuzzer.]), [], [enable_libfuzzer=no])
@@ -236,7 +251,7 @@ AC_MSG_NOTICE([Multithreading: $enable_multithreading])
 AC_MSG_NOTICE([Symbol visibility: $enable_visibility])
 AC_MSG_NOTICE([libaom decoder: $have_aom])
 AC_MSG_NOTICE([libaom encoder: $have_aom])
-AC_MSG_NOTICE([rav1e encoder: $enable_local_rav1e])
+AC_MSG_NOTICE([rav1e encoder: $have_rav1e (local: $enable_local_rav1e)])
 AC_MSG_NOTICE([libde265 decoder: $have_libde265])
 AC_MSG_NOTICE([libx265 encoder: $have_x265])
 AC_MSG_NOTICE([JPEG output: $have_libjpeg])
diff --git a/libheif/Makefile.am b/libheif/Makefile.am
index a2b91ea..f17fb39 100644
--- a/libheif/Makefile.am
+++ b/libheif/Makefile.am
@@ -21,8 +21,8 @@ if HAVE_X265
 ADDITIONAL_LIBS += $(x265_LIBS)
 endif
 
-if ENABLE_LOCAL_RAV1E
-ADDITIONAL_LIBS += ../third-party/rav1e/target/release/librav1e.a
+if HAVE_RAV1E
+ADDITIONAL_LIBS += $(rav1e_LIBS)
 endif
 
 libheif_la_CPPFLAGS =
@@ -34,13 +34,10 @@ libheif_la_CXXFLAGS = \
   $(aom_CFLAGS) \
   $(libde265_CFLAGS) \
   $(x265_CFLAGS) \
+  $(rav1e_CFLAGS) \
   -DLIBHEIF_EXPORTS -I$(top_srcdir)
 libheif_la_LIBADD = $(ADDITIONAL_LIBS)
 
-if ENABLE_LOCAL_RAV1E
-libheif_la_CXXFLAGS += -I../third-party/rav1e/target/release/include
-endif
-
 libheif_la_LDFLAGS = -version-info $(LIBHEIF_CURRENT):$(LIBHEIF_REVISION):$(LIBHEIF_AGE)
 if MINGW
 libheif_la_LDFLAGS += -no-undefined
@@ -91,7 +88,7 @@ libheif_la_SOURCES += \
   heif_decoder_libde265.h
 endif
 
-if ENABLE_LOCAL_RAV1E
+if HAVE_RAV1E
 libheif_la_SOURCES += \
   heif_encoder_rav1e.cc \
   heif_encoder_rav1e.h
