diff --git a/configure.ac b/configure.ac
index fb2a23b..15715da 100644
--- a/configure.ac
+++ b/configure.ac
@@ -99,6 +99,22 @@ PKG_CHECK_MODULES([aom], [aom], [
 AM_CONDITIONAL([HAVE_AOM], [test "x$have_aom" = "xyes"])
 AC_SUBST([have_aom])
 
+if eval "test x$have_aom = xyes"; then
+AC_CHECK_HEADER([aom/aom_decoder.h], [
+    AC_DEFINE([HAVE_AOM_DECODER], [1], [Whether aom decoder was found.])
+    have_aom_decoder="yes"
+], [have_aom_decoder="no"])
+AM_CONDITIONAL([HAVE_AOM_DECODER], [test "x$have_aom_decoder" = "xyes"])
+AC_SUBST([have_aom_decoder])
+
+AC_CHECK_HEADER([aom/aom_encoder.h], [
+    AC_DEFINE([HAVE_AOM_ENCODER], [1], [Whether aom encoder was found.])
+    have_aom_encoder="yes"
+], [have_aom_encoder="no"])
+AM_CONDITIONAL([HAVE_AOM_ENCODER], [test "x$have_aom_encoder" = "xyes"])
+AC_SUBST([have_aom_encoder])
+fi
+
 PKG_CHECK_MODULES([rav1e], [rav1e], [
     AC_DEFINE([HAVE_RAV1E], [1], [Whether rav1e was found.])
     AC_SUBST(rav1e_CFLAGS)
@@ -249,8 +265,8 @@ AC_C_CXX_COMPILE_FLAGS([-Wno-error=potentially-evaluated-expression])
 AC_MSG_NOTICE([---------------------------------------])
 AC_MSG_NOTICE([Multithreading: $enable_multithreading])
 AC_MSG_NOTICE([Symbol visibility: $enable_visibility])
-AC_MSG_NOTICE([libaom decoder: $have_aom])
-AC_MSG_NOTICE([libaom encoder: $have_aom])
+AC_MSG_NOTICE([libaom decoder: $have_aom_decoder])
+AC_MSG_NOTICE([libaom encoder: $have_aom_encoder])
 AC_MSG_NOTICE([rav1e encoder: $have_rav1e (local: $enable_local_rav1e)])
 AC_MSG_NOTICE([libde265 decoder: $have_libde265])
 AC_MSG_NOTICE([libx265 encoder: $have_x265])
diff --git a/libheif.pc.in b/libheif.pc.in
index 1fa90c4..ce345bf 100644
--- a/libheif.pc.in
+++ b/libheif.pc.in
@@ -4,8 +4,8 @@ libdir=@libdir@
 includedir=@includedir@
 builtin_h265_decoder=@have_libde265@
 builtin_h265_encoder=@have_x265@
-builtin_avif_decoder=@have_aom@
-builtin_avif_encoder=@have_aom@
+builtin_avif_decoder=@have_aom_decoder@
+builtin_avif_encoder=@have_aom_encoder@
 
 Name: libheif
 Description: HEIF image codec.
diff --git a/libheif/Makefile.am b/libheif/Makefile.am
index f17fb39..3f2410a 100644
--- a/libheif/Makefile.am
+++ b/libheif/Makefile.am
@@ -74,14 +74,18 @@ libheif_la_SOURCES = \
   nclx.cc \
   nclx.h
 
-if HAVE_AOM
+if HAVE_AOM_DECODER
 libheif_la_SOURCES += \
-  heif_encoder_aom.cc \
-  heif_encoder_aom.h \
   heif_decoder_aom.cc \
   heif_decoder_aom.h
 endif
 
+if HAVE_AOM_ENCODER
+libheif_la_SOURCES += \
+  heif_encoder_aom.cc \
+  heif_encoder_aom.h
+endif
+
 if HAVE_LIBDE265
 libheif_la_SOURCES += \
   heif_decoder_libde265.cc \
diff --git a/libheif/heif_plugin_registry.cc b/libheif/heif_plugin_registry.cc
index 126aa06..c62cfc0 100644
--- a/libheif/heif_plugin_registry.cc
+++ b/libheif/heif_plugin_registry.cc
@@ -35,8 +35,11 @@
 #include "heif_encoder_x265.h"
 #endif
 
-#if HAVE_AOM
+#if HAVE_AOM_ENCODER
 #include "heif_encoder_aom.h"
+#endif
+
+#if HAVE_AOM_DECODER
 #include "heif_decoder_aom.h"
 #endif
 
@@ -78,8 +81,11 @@ public:
     heif::register_encoder(get_encoder_plugin_x265());
 #endif
 
-#if HAVE_AOM
+#if HAVE_AOM_ENCODER
     heif::register_encoder(get_encoder_plugin_aom());
+#endif
+
+#if HAVE_AOM_DECODER
     heif::register_decoder(get_decoder_plugin_aom());
 #endif
 
